File: 0015-maximum-depth-of-n-ary-tree/solution.swift

public class Node {
    public var val: Int
    public var children: [Node]
    public init(_ val: Int) {
        self.val = val
        self.children = []
    }
}


func maxDepth(_ root: Node?) -> Int {
    guard let root = root else { return 0 }
    
    var depth = 0
    
    for child in root.children {
        depth = max(depth, maxDepth(child))
    }
    
    return depth + 1
}

File: 0015-maximum-depth-of-n-ary-tree/solution.go
package main

type Node struct {
	Val      int
	Children []*Node
}

func maxDepth(root *Node) int {
	if root == nil {
		return 0
	}

	depth := 0

	for _, child := range root.Children {
		depth = max(depth, maxDepth(child))
	}

	return 1 + depth
}
